### PR TITLE
Add MediaQuery to View

### LINKED ISSUE #115375: Split MediaQuery into ViewQuery and PlatformQuery
MediaQuery mixes platform-specific concepts (that are shared among all views) with view-specific concepts. This makes MediaQuery cumbersome to use in a multi view environment: The View widget needs to inject the view-specific data into the widget tree, without overriding the platform-specific data from the surrounding context. This can be done with the existing MediaQuery concept, but it would require copying the platform-specific data from the context and makes rebuilds less targeted when the data changes. 

Since we are removing single view assumptions elsewhere in the framework as well, it would be cleaner to split MediaQuery into ViewQuery (which contains the view-specific data) and PlatformQuery (which contains the rest). The data from the PlatformQuery would be injected near the root of the tree and is accessible from all views. The data from the ViewQuery would be injected by the View widget and its only available within that View.

If we implement the InheritanceBoundary (https://github.com/flutter/flutter/issues/115381), the PlatformQuery would not be an inheritance boundary, while ViewQuery would be one.

Data moving to ViewQuery:
* devicePixelRatio
* displayFeatures
* orientation
* padding
* size
* systemGestureInsets
* viewInsets
* viewPadding

Data moving to PlatformQuery:
* accessibleNavigation
* alwaysUse24HourFormat 
* boldText
* disableAnimations
* gestureSettings
* highContrast
* invertColors
* navigationMode
* platformBrightness
* textScaleFactor

Migration: We would deprecate MediaQuery and redirect people to PlatformQuery and ViewQuery. During the deprecation policy, MediaQuery would be reimplemented by composing PlatformQuery and ViewQuery internally.

### PR DESCRIPTION
**Pull request by goderbauer**
Part of https://github.com/flutter/flutter/issues/116929
Fixes https://github.com/flutter/flutter/issues/115375

### CODE REVIEW

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 1]
```diff
@@ -183,24 +183,54 @@ class MediaQueryData {
...
+  factory MediaQueryData.fromWindow(ui.FlutterView window) => MediaQueryData.fromView(window);
+
+  /// Creates data for a [MediaQuery] based on the given `view`.
+  ///
+  /// If provided, the `platformData` is used to fill in the platform-specific
+  /// aspects of the newly created [MediaQueryData]. If `platformData` is null,
+  /// the `view`'s [PlatformDispatcher] is consulted to construct the
+  /// platform-specific data.
+  ///
+  /// Data, that is exposed directly on the [FlutterView] is considered
```
Comment (gspencergoog) [inline packages/flutter/lib/src/widgets/media_query.dart:195]:
```suggestion
  /// Data which is exposed directly on the [FlutterView] is considered
```

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 2]
```diff
@@ -183,24 +183,54 @@ class MediaQueryData {
...
+  /// view-specific. Data, that is only exposed via the
```
Comment (gspencergoog) [inline packages/flutter/lib/src/widgets/media_query.dart:196]:
```suggestion
  /// view-specific. Data which is only exposed via the
```

File: packages/flutter/lib/src/widgets/view.dart [hunk 3]
```diff
@@ -92,3 +107,12 @@ class View extends InheritedWidget {
     return result!;
   }
 }
+
+class _ViewScope extends InheritedWidget {
```
Comment (gspencergoog) [inline packages/flutter/lib/src/widgets/view.dart:111]: I call these kinds of private inherited widgets "markers" (e.g. `_FocusMarker`, `_ActionsMarker`, etc.). As a name, "scope" works too, but you might consider using "marker" for consistency.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: packages/flutter/lib/src/widgets/view.dart (hunk 3 above)
Comment (goderbauer) [reply packages/flutter/lib/src/widgets/view.dart:111]: I did a quick count: Maybe we should rename those `Marker`s to `Scope`s. Currently, we have 5 widgets in the code base that use the suffix Marker, and 13 that uses Scope.

Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/view.dart:111]: Works for me.

**Comment by goderbauer** [comment]
The"google testing" failures were caused by flaws in some internal tests that this change exposed. Fixes for the tests have been submitted for review in cl/504607930, cl/504609793, and cl/504611321.

**Comment by zanderso** [comment]
DBC

SkiaPerf is reporting that this change increases the release (compressed) size of hello world by a little under 9KB, which is surprising to me given the size of this PR, so I'm wondering if it somehow made tree shaking harder?

https://flutter-flutter-perf.skia.org/e/?queries=sub_result%3Dapp_framework_uncompressed_bytes%26sub_result%3Drelease_size_bytes%26test%3Dhello_world_ios__compile&selected=commit%3D33030%26name%3D%252Carch%253Darm%252Cbranch%253Dmaster%252Cconfig%253Ddefault%252Cdevice_type%253DiPhone_11%252Cdevice_version%253Dnone%252Chost_type%253Dmac%252Csub_result%253Drelease_size_bytes%252Ctest%253Dhello_world_ios__compile%252C

**Comment by goderbauer** [comment]
Discussed on chat: This adds a MediaQuery for the first time to the hello world app, which explains the size increase there. Any real world app will already contain a MediaQuery so it shouldn't be affected at all (as confirmed by looking at the numbers for Gallery).

**Pull request merged**